pkg/alertmanager/amcfg.go

```diff
@@ -153,11 +153,19 @@ func (cb *configBuilder) initializeFromAlertmanagerConfig(ctx context.Context, g
 	}
 
 	for _, muteTimeInterval := range amConfig.Spec.MuteTimeIntervals {
-		mti, err := convertMuteTimeInterval(&muteTimeInterval, crKey)
+		ti, err := convertTimeInterval(&muteTimeInterval, crKey)
 		if err != nil {
 			return err
 		}
-		globalAlertmanagerConfig.MuteTimeIntervals = append(globalAlertmanagerConfig.MuteTimeIntervals, mti)
+		globalAlertmanagerConfig.TimeIntervals = append(globalAlertmanagerConfig.TimeIntervals, ti)
```

Comment:
The code is simply moving away from MuteTimeIntervals to TimeInterval but shouldn't MuteTimeInterval be the default and just use TimeInterval when the AM version is >= v0.24? Or am I missing something?

Reply:
Sorry, indeed it was a replacement. Fixed to support both syntax